content-services/content-sources-backend
# Fixes 4110,4209: set gpg & modular hotfixes in candlepin

## Summary

Adds support for gpg URL (On Content) & Modular hotfixes (as an override) in candlepin
Adds an update repository task to update these settings

## Testing steps

1.  Register a system following https://github.com/content-services/content-sources-backend/blob/main/docs/register_client.md
3. Create a custom repository with no GPG key and module filtering enabled (defaults)
4. Create a content template with the repository
5. associate the system to the template
6. On the client,  run   `subscription-manager refresh && subscription-manager repo-override && yum repolist`  check the /etc/yum.repos.d/redhat.repo file.  Verify that gpgcheck=0  and module_hotfixes=1 is not present.
7. Update the repository, add a GPG Key, toggle modularity filtering
8. On the client, run   `subscription-manager refresh && subscription-manager repo-override && yum repolist`  check the /etc/yum.repos.d/redhat.repo file.  Verify that gpgcheck=1, gpgkey=$URL  and module_hotfixes=1 is present.
9. Undo 7, and redo 8, and verify the settings reverted back to their normal behavior.
## Checklist
